### ISSUE TITLE
Occasional "Broken Pipe" diagnostic from t/op/pwent.t

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#282]
Migrated from [rt.perl.org#1104](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=1104) (status was 'resolved')

Searchable as RT1104$

**Comment by p5pRT** [comment]
### From mooring@antares.Tymnet.COM



t/op/pwent\.t generates an occasional "Broken Pipe" diagnostic from
the shell that is invoked by opening a pipe to ypcat\. The problem
is that pwent\.t closes the reader end of the pipe after $max
lines\, and its possible for ypcat to get a SIGPIPE on the next
write\, thus inducing the shell that started it to send the diagnostic\.

It's utterly harmless\, aside from introducing some alarming\-looking
output during 'make test'\.

The tiny patch below fixes it for me&#8203;:


\*\*\* t/op/pwent\.t\.dist    Wed Jul 28 23&#8203;:13&#8203;:27 1999
\-\-\- t/op/pwent\.t    Wed Jul 28 17&#8203;:09&#8203;:36 1999
\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*
\*\*\* 79\,85 \*\*\*\*
&nbsp;      warn "\# Your $where line $\. is empty\.\\n";
&nbsp;      next;
&nbsp;      \}
\!     last if $n == $max;
&nbsp;      \# In principle we could whine if @&#8203;s \!= 7 but do we know enough
&nbsp;      \# of passwd file formats everywhere?
&nbsp;      if \(@&#8203;s == 7\) \{
\-\-\- 79\,85 \-\-\-\-
&nbsp;      warn "\# Your $where line $\. is empty\.\\n";
&nbsp;      next;
&nbsp;      \}
\!     next if $n == $max;
&nbsp;      \# In principle we could whine if @&#8203;s \!= 7 but do we know enough
&nbsp;      \# of passwd file formats everywhere?
&nbsp;      if \(@&#8203;s == 7\) \{

<details><summary>Perl Info</summary>

```


Site configuration information for perl 5.00503:

Configured by mooring at Tue Mar 30 16:13:23 PST 1999.

Summary of my perl5 (5.0 patchlevel 5 subversion 3) configuration:
  Platform:
    osname=solaris, osvers=2.6, archname=sun4-solaris
    uname='sunos grimoire 5.6 generic sun4u sparc sunw,ultra-2 '
    hint=previous, useposix=true, d_sigaction=define
    usethreads=undef useperlio=undef d_sfio=undef
  Compiler:
    cc='gcc -B/usr/ccs/bin/', optimize='-O3 -g', gccversion=egcs-2.91.66 19990314 (egcs-1.1.2 release)
    cppflags='-DDEBUGGING -I/usr/local/include -I/opt/local/include -I/opt/gnu/include'
    ccflags ='-DDEBUGGING -I/usr/local/include -I/opt/local/include -I/opt/gnu/include'
    stdchar='unsigned char', d_stdstdio=define, usevfork=false
    intsize=4, longsize=4, ptrsize=4, doublesize=8
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=16
    alignbytes=8, usemymalloc=y, prototype=define
  Linker and Libraries:
    ld='gcc -B/usr/ccs/bin/', ldflags =' -L/usr/local/lib -L/opt/local/lib -L/opt/gnu/lib'
    libpth=/usr/local/lib /opt/local/lib /opt/gnu/lib /lib /usr/lib /usr/ccs/lib
    libs=-lsocket -lnsl -lgdbm -ldb -ldl -lm -lc -lcrypt
    libc=/lib/libc.so, so=so, useshrplib=false, libperl=libperl.a
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags=' '
    cccdlflags='-fPIC', lddlflags='-G -L/usr/local/lib -L/opt/local/lib -L/opt/gnu/lib'

Locally applied patches:
    


@INC for perl 5.00503:
    /opt/local/lib/perl5/5.00503/sun4-solaris
    /opt/local/lib/perl5/5.00503
    /opt/local/lib/perl5/site_perl/5.005/sun4-solaris
    /opt/local/lib/perl5/site_perl/5.005
    .


Environment for perl 5.00503:
    HOME=/home/grimoire/a/mooring
    LANG (unset)
    LANGUAGE (unset)
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/opt/egcs/bin:.:/home/grimoire/a/mooring/bin:/opt/gnu/bin:/usr/local/bin:/usr/ucb:/usr/bin:/usr/sbin:/sbin:/bin:/usr/ccs/bin:/etc:/usr/etc
    PERL_BADLANG (unset)
    SHELL=/bin/csh

```
</details>

**Comment by p5pRT** [comment]
### From @jhi


Ed Mooring writes&#8203;:
 > 
 > This is a bug report for perl from mooring@&#8203;grimoire\.tymnet\.com\,
 > generated with the help of perlbug 1\.26 running under perl 5\.00503\.
 > 
 > 
 > \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
 > \[Please enter your report here\]
 > 
 > t/op/pwent\.t generates an occasional "Broken Pipe" diagnostic from
 > the shell that is invoked by opening a pipe to ypcat\. The problem
 > is that pwent\.t closes the reader end of the pipe after $max
 > lines\, and its possible for ypcat to get a SIGPIPE on the next
 > write\, thus inducing the shell that started it to send the diagnostic\.
 > 
 > It's utterly harmless\, aside from introducing some alarming\-looking
 > output during 'make test'\.
 > 
 > The tiny patch below fixes it for me&#8203;:

Thanks\.

An alternative patch has already been submitted for this problem where
instead of keeping on reading the whole remaining input is slurped in
and then we last; \(this works faster for sites with umpteen thousand
accounts\)

\-\- 
$jhi\+\+; \# http&#8203;://www\.iki\.fi/jhi/
&nbsp;        \# There is this special biologist word we use for 'stable'\.
&nbsp;        \# It is 'dead'\. \-\- Jack Cohen
